tc/sgx/trusted_worker_manager/enclave/kme_workload/kme_workload_plug-in.cpp

```diff
@@ -59,10 +82,91 @@ void KMEWorkloadProcessor::GetUniqueId(
     const std::vector<tcf::WorkOrderData>& in_work_order_data,
-    std::vector<tcf::WorkOrderData>& out_work_order_data,
-    ExtWorkOrderInfoKME* ext_wo_info_kme) {
+    std::vector<tcf::WorkOrderData>& out_work_order_data) {
+    // If in_work_order_data is empty
+    if (in_work_order_data.size() == 0) {
+        this->SetStatus((int)ERR_WPE_REG_FAILED,
+            out_work_order_data);
+        ThrowIf<ValueError>(true, "Registration request is empty");
+    }
+    /* Registration request is serialized json string
+      {
+       "attestation_data": <attestation_report>
+      }
+    */
+    ByteArray reg_request = in_work_order_data[0].decrypted_data;
+
+    std::string reg_data_string = ByteArrayToString(reg_request);
+    /// Parse the registration request
+    JsonValue reg_data_string_parsed(json_parse_string(
+        reg_data_string.c_str()));
+    if (reg_data_string_parsed.value == nullptr) {
+        this->SetStatus((int)ERR_WPE_REG_FAILED,
+            out_work_order_data);
+        ThrowIf<ValueError>(true,
+            "Failed to parse the WPE registration request");
+    }
+    JSON_Object* reg_data_json_object = \
+        json_value_get_object(reg_data_string_parsed);
+    if (reg_data_json_object == nullptr) {
+        this->SetStatus((int)ERR_WPE_REG_FAILED,
+            out_work_order_data);
+        ThrowIf<ValueError>(true,
+            "Invalid registration request, expecting valid json object");
+    }
 
-    // To be implemented
+    const char* s_value = nullptr;
+    
+    s_value = json_object_dotget_string(
+        reg_data_json_object, "attestation_data");
+    if (s_value == nullptr) {
+        this->SetStatus((int)ERR_WPE_REG_FAILED,
+            out_work_order_data);
+        ThrowIf<ValueError>(true, "Invalid attestation data");
+    }
+
+    std::string attestation_data_str(s_value);
+    ByteArray attestation_data(attestation_data_str.begin(),
+        attestation_data_str.end());
+
+    ByteArray e_key = {};
+    ByteArray verification_key_hash = {};
+    ByteArray mr_enclave = {};
+    ByteArray mr_signer = {};
+    int err = this->ext_wo_info_kme->VerifyAttestation(
+        attestation_data, mr_enclave, mr_signer,
+        verification_key_hash, e_key);
+    if (err != 0) {
+        this->SetStatus(err, out_work_order_data);
+        ThrowIf<ValueError>(true, "WPE attestation verification failed");
+    }
+    auto search = this->sig_key_map.find(verification_key_hash);
+    if (search != this->sig_key_map.end()) {
+        /// Compare MRENCLAVE value
+        EnclaveData* enclaveData = EnclaveData::getInstance();
+        std::string ext_data(enclaveData->get_extended_data());
+        std::string mr_enclave_str = ByteArrayToString(mr_enclave);
+        if (ext_data.compare(mr_enclave_str) != 0) {
+            this->SetStatus((int)ERR_MRENCLAVE_NOT_MATCH,
+                out_work_order_data);
+            ThrowIf<ValueError>(true, "WPE MRENCLAVE value didn't match");
+        }
```

Comment:
Need to check if WPE is already registered with KME using this unique_key (verification_key) generated by KME